xarray/tests/test_backends.py

```diff
@@ -957,6 +988,37 @@ def roundtrip(self, data, save_kwargs={}, open_kwargs={},
                               autoclose=self.autoclose, **open_kwargs) as ds:
                 yield ds
 
+    @contextlib.contextmanager
+    def roundtrip_append(self, data, save_kwargs={}, open_kwargs={},
+                         allow_cleanup_failure=False):
+        with create_tmp_file(
+                allow_cleanup_failure=allow_cleanup_failure) as tmp_file:
+            for i, key in enumerate(data.variables):
+                mode = 'a' if i > 0 else 'w'
+                data[[key]].to_netcdf(tmp_file, mode=mode, engine='scipy',
+                                      **save_kwargs)
+                data.close()
+            with open_dataset(tmp_file,
+                              engine='scipy', **open_kwargs) as ds:
+                yield ds
+
+    def test_append_write(self):
+        # regression for GH1215
+        data = create_test_data()
+        with self.roundtrip_append(data) as actual:
+            assert_allclose(data, actual)
+
+    def test_append_overwrite_values(self):
+        data = create_test_data()
+        with create_tmp_file(allow_cleanup_failure=False) as tmp_file:
+            data.to_netcdf(tmp_file, mode='w', engine='scipy')
+            data['var2'][:] = -999
+            data['var9'] = data['var2'] * 3
+            data[['var2', 'var9']].to_netcdf(tmp_file, mode='a',
+                                             engine='scipy')
+            actual = open_dataset(tmp_file, engine='scipy')
```

Comment:
This is triggering a test failure on Windows, where you apparently can't open the same file twice.